jovian/callbacks/jovian_fastai_callback.py

```diff
@@ -0,0 +1,54 @@
+from fastai.callback import Callback
+from fastai.basic_train import Learner
+from jovian import log_hyperparams, log_metrics
+
+
+class JovianFastaiCallback(Callback):
+    """FastAI callback to log hyperparameters and metrics during model training.
+
+    Arguments:
+        learn (Learner): A learner object with which you're fitting your model
+
+        arch_name (string): A name for the architecture that you're using 
+    """
+
+    def __init__(self, learn: Learner, arch_name):
+        self.learn = learn
+        self.arch_name = arch_name
+        self.hyp_names = ['arch_name', 'epochs', 'batch_size',
+                          'loss_func', 'opt_func', 'weight_decay', 'learning_rate']
+        self.met_names = ['epoch', 'train_loss']
+        # existance of validation dataset
+        self.valid_set = self.learn.data.valid_dl.items.any()
+        if(self.valid_set):
+            self.met_names.append('valid_loss')
+
+    def on_train_begin(self, **ka):
+
+        hyp_values = [self.arch_name,
```

Comment:
would be better to create dict object directly here. Other we will have to be very careful not to change the ordering of the hyp_names. You can use OrderedDict for ordered dictionary.

Reply:
Now I'm initializing the hyp dictionary here. Is OrderedDict required as i have observed that the front-end sort the keys and then displays?

Reply:
OrderedDict won't be required.